tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -982,6 +982,14 @@ class SemanticAnnotator : public SourceEntityWalker {
 
     unsigned ByteOffset = SM.getLocOffsetInBuffer(Range.getStart(), BufferID);
     unsigned Length = Range.getByteLength();
+    if (D == D->getASTContext().getOptionalNoneDecl() && Length == 3) {
+      // If a 'nil' literal occurs in a swift-case statement, it gets replaced
+      // by a reference to 'Optional.none' in the AST. We want to continue
+      // highlighting 'nil' as a keyword and not as an enum element.
+      // Detect this case if 'none' was spelled with 3 characters, which
+      // indicates that it was rewritten from 'nil'.
```

Comment:
`Length == 3` might be enough in real world, but `SM.extractText(Range) == "nil"` is more explicit, safe, and not so heavy. If we don't have strong reason not to, could you check the actual spelling?

Reply:
Also, other "deciding annotate or not" checks are in `visitDeclReference()` instead of `annote()`. Could you move this logic to it?

Reply:
Good suggestions, thanks. I though I would have needed to go to the lexer to get the original source text but of course I don’t.